recipe/build-arrow.sh

```diff
@@ -60,6 +60,9 @@ fi
 if [[ "${target_platform}" == "linux-ppc64le" ]]; then
   EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_ALTIVEC=ON"
 fi
+if [[ "${target_platform}" != "linux-s390x" ]]; then
+  EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_USE_LD_GOLD=ON"
```

Comment:
With ld.gold, I see an issue on s390x. 
```
_build_env/bin/../lib/gcc/s390x-conda-linux-gnu/7.5.0/../../../../s390x-conda-linux-gnu/bin/ld.gold: error: cannot open XXXX/_build_env/bin/../s390x-conda-linux-gnu/sysroot/lib/ld64.so.1: No such file or directory
```

```
This is seen when I looked into the directory with ls -l in the build environment
XXXX/_build_env/bin/../s390x-conda-linux-gnu/sysroot/lib/ld64.so* -> ld-2.17.so
```

So to avoid the error, I marked it for `not s390x`.